src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs

```diff
@@ -56,8 +57,8 @@ public override void ExecuteCmdlet()
                 {
                     if (this.BackupStorageRedundancy.HasValue)
                     {
-                        BackupStorageConfig vaultStorageRequest = new BackupStorageConfig();
-                        vaultStorageRequest.StorageModelType = BackupStorageRedundancy.ToString();
+                        BackupResourceConfigResource vaultStorageRequest = new BackupResourceConfigResource();
```

Comment:
Please try to use BackupResourceConfig as the replacement for BackupStorageConfig being used earlier

Reply:
This method from the backup sdk is expecting BackupResourceConfigResource as one of the parameters GetRecoveryServicesBackupClient.BackupResourceStorageConfigs.UpdateWithHttpMessagesAsync(
                resouceGroupName, vaultName, backupStorageConfig, GetRequestHeaders());

Reply:
Okay, your changes look fine